ext/socket/ipsocket.c

```diff
@@ -167,10 +167,930 @@ init_inetsock_internal(VALUE v)
+                    wait_arg.writefds = &writefds;
+
+                    set_timeout_tv(&connection_attempt_delay_strage, 250, now);
+                    connection_attempt_delay_expires_at = &connection_attempt_delay_strage;
+
+                    if (!any_addrinfos(&resolution_store)) {
+                        if (NIL_P(connect_timeout)) {
+                            user_specified_connect_timeout_storage = (struct timeval){ -1, -1 };
+                        } else {
+                            struct timeval connect_timeout_tv = rb_time_interval(connect_timeout);
+                            user_specified_connect_timeout_storage = add_ts_to_tv(connect_timeout_tv, now);
+                        }
+                        user_specified_connect_timeout_at = &user_specified_connect_timeout_storage;
+                    }
+
+                    if (debug) {
+                        printf("[DEBUG] %d: connection attempt fd size %d\n", count, arg->connection_attempt_fds_size);
+                        for (int i = 0; i < arg->connection_attempt_fds_size; i++) {
+                            printf("[DEBUG] %d: connection attempt fd %d\n", count, arg->connection_attempt_fds[i]);
+                        }
+                    }
+                    break;
+                }
+
+                if (debug) printf("[DEBUG] %d: connection failed\n", count);
+                last_error = errno;
+                close(fd);
+
+                if (any_addrinfos(&resolution_store)) continue;
+                if (in_progress_fds(arg->connection_attempt_fds, arg->connection_attempt_fds_size)) break;
+                if (!resolution_store.is_all_finised) break;
+
+                if (local_status < 0) {
+                    rsock_syserr_fail_host_port(last_error, syscall, arg->local.host, arg->local.serv);
+                }
+                rsock_syserr_fail_host_port(last_error, syscall, arg->remote.host, arg->remote.serv);
+            }
+        }
+
+        if (connected_fd >= 0) break;
+
+        if (debug) printf("[DEBUG] %d: resolution_delay_expires_at %p\n", count, resolution_delay_expires_at);
+        if (debug) printf("[DEBUG] %d: connection_attempt_delay_expires_at %p\n", count, connection_attempt_delay_expires_at);
+        if (debug) printf("[DEBUG] %d: user_specified_resolv_timeout_at %p\n", count, user_specified_resolv_timeout_at);
+        if (debug) printf("[DEBUG] %d: user_specified_connect_timeout_at %p\n", count, user_specified_connect_timeout_at);
+        ends_at = select_expires_at(
+            &resolution_store,
+            resolution_delay_expires_at,
+            connection_attempt_delay_expires_at,
+            user_specified_resolv_timeout_at,
+            user_specified_connect_timeout_at
+        );
+        if (ends_at) {
+            if (debug) printf("[DEBUG] %d: ends_at->tv_sec %ld\n", count, ends_at->tv_sec);
+            if (debug) printf("[DEBUG] %d: ends_at->tv_usec %d\n", count, ends_at->tv_usec);
```

Comment:
32bit is enough for 1,000,000, and ruby assumes that `int` has 32bit at least.

```suggestion
            if (debug) printf("[DEBUG] %d: ends_at->tv_sec %"PRI_TIMET_PREFIX"d\n", count, ends_at->tv_sec);
            if (debug) printf("[DEBUG] %d: ends_at->tv_usec %d\n", count, (int)ends_at->tv_usec);
```

Reply:
Thank you for your comment. `If (debug)` will be removed after CI passes.